extensions/amp-analytics/0.1/transport.js

```diff
@@ -131,7 +133,10 @@ export class Transport {
         typeof image == 'object' && image['suppressWarnings'];
       Transport.sendRequestUsingImage(
         this.win_,
-        getRequest(false),
+        // If xhrpost or beacon failed, don't append extraUrlParams to url
+        xhrpost || beacon
+          ? {'url': getRequest(this.useBody_).url}
+          : getRequest(false),
```

Comment:
Hmmm would this break the use case when the extraUrlParams is not an object?

Reply:
What do you mean by not an object? Isn't `extraUrlParams` always an object?